rust_src/src/buffers.rs

```diff
@@ -1153,10 +1154,16 @@ pub fn generate_new_buffer_name(name: LispStringRef, ignore: LispObject) -> Lisp
     }
 }
 
+pub unsafe fn per_buffer_idx_from_field_offset(
+    offset: FieldOffset<Lisp_Buffer, LispObject>,
+) -> isize {
+    unsafe { *offset.apply_ptr_mut(&mut buffer_local_flags) }.to_fixnum_unchecked() as isize
```

Comment:
can't we use `per_buffer_idx` here ?

Reply:
With effort. The key issue is the difference in pointer math in Rust versus C.

In C, the idiom is `foo = ptr + num_bytes;` but in Rust the code looks like this `foo = ptr.offset(num_things);`. The difference is the C math is done in raw bytes. The Rust equivalent in C looks like this `PtrType *foo = ptr + (num_things * sizeof(PtrType));`.

Despite the variable name `offset`, it should really be `count`. I will update that in my refactor PR. I forgot last night.

The `Lisp_Fwd` type in Rust uses the `FieldOffset` type so the `apply_ptr` method is more idiomatic. Otherwise we would reimplement it by using `get_byte_offset()` and dividing it by size of `LispObject` to pass to `ptr.offset()`.

Reply:
Got it.